tests/keras/optimizers_test.py

```diff
@@ -63,64 +63,84 @@ def _test_optimizer(optimizer, target=0.75):
     assert_allclose(kernel, 1.)
     assert_allclose(bias, 2.)
 
+@keras_test
+def _test_no_grad(optimizer):
```

Comment:
I would say this is only needed once (e.g. for SGD) since this is code that is shared across all optimizers.